src/es/components/organisms/registerForm/RegisterForm.js

```diff
@@ -257,6 +257,109 @@ export default class RegisterForm extends Shadow() {
+            elem.value = ""
+            e.target.setCustomValidity(validityMessage);
+            e.target.reportValidity();
+          }
+        }
+      })
+    })
+
+    // pattern right format validation
+    const patternTextInputFields = this.root.querySelectorAll('input[type="text"][custom-pattern-validation]')
+    Array.from(patternTextInputFields).forEach(elem => {
+      const validityMessage = elem.getAttribute("custom-validity-message")
+
+      elem.addEventListener("keydown", (e) => {
+        // ignore NaN e.keys except C,H,E, left and right arrows and spec chars that needed to type the code
+        if ((isNaN(+e.key) || +e.keyCode === 69 || +e.keyCode === 107) && +e.keyCode !== 8
+          && (isNaN(+e.key) || +e.keyCode === 69 || +e.keyCode === 107) && +e.keyCode !== 16
+          && (isNaN(+e.key) || +e.keyCode === 69 || +e.keyCode === 107) && +e.keyCode !== 37
+          && (isNaN(+e.key) || +e.keyCode === 69 || +e.keyCode === 107) && +e.keyCode !== 39
+          && (isNaN(+e.key) || +e.keyCode === 69 || +e.keyCode === 107) && +e.keyCode !== 67
+          && (isNaN(+e.key) || +e.keyCode === 69 || +e.keyCode === 107) && +e.keyCode !== 72
+          && (isNaN(+e.key) || +e.keyCode === 69 || +e.keyCode === 107) && +e.keyCode !== 69
+          && (isNaN(+e.key) || +e.keyCode === 69 || +e.keyCode === 107) && +e.keyCode !== 109
+          && (isNaN(+e.key) || +e.keyCode === 69 || +e.keyCode === 107) && +e.keyCode !== 190
+          && (isNaN(+e.key) || +e.keyCode === 69 || +e.keyCode === 107) && +e.keyCode !== 110) {
+          e.target.setCustomValidity(validityMessage);
+          e.target.reportValidity();
+          e.preventDefault();
+        }
+      })
+
+      elem.addEventListener("keyup", (e) => {
+        const currentValue = elem.value
+        if (
+          currentValue.substring(0, 3) === "CHE"
+          && currentValue.split('.').length === 3
+          && currentValue.split('.')[1].split('').every(e => !isNaN(+e))
+          && currentValue.split('.')[2].split('').every(e => !isNaN(+e))
+          && currentValue.length === 15
+        ) {
+          e.target.setCustomValidity("");
+          e.target.reportValidity();
+        } else {
+          e.target.setCustomValidity(validityMessage);
+          e.target.reportValidity();
+        }
+      })
+
+      elem.addEventListener("change", (e) => {
+        const currentValue = elem.value
+        if (
+          currentValue.substring(0, 3) === "CHE"
+          && currentValue.split('.').length === 3
+          && currentValue.split('.')[1].split('').every(e => !isNaN(+e))
+          && currentValue.split('.')[2].split('').every(e => !isNaN(+e))
+          && currentValue.length === 15
+        ) {
+          e.target.setCustomValidity("");
+          e.target.reportValidity();
+        } else {
+          e.target.setCustomValidity(validityMessage);
+          e.target.reportValidity();
+          elem.value = ""
+        }
+      })
+    })
```

Comment:
Where do these requirements come from? I don't understand why the frontend should intercept this. Is the HTML validation using the pattern attribute not sufficient?

Reply:
Yes, good. Thank you for the clarification.